lib/Page.js

```diff
@@ -125,6 +126,11 @@ class Page extends EventEmitter {
     this.emit('error', new Error('Page crashed!'));
   }
 
+  _onLogEntryAdded(event) {
+    const {level, text, args} = event.entry;
+    this.emit(Page.Events.Console, new ConsoleMessage(level, text, args));
```

Comment:
args are [`RemoteObjects`](https://vanilla.aslushnikov.com/#Log.LogEntry) here; you'd need to convert these to JSHandles first.

It would be ideal if you can reuse all the logic that's already in [`page._onConsoleAPI`](https://github.com/GoogleChrome/puppeteer/blob/0820d48f804ce924af85fe160ea20a210f734046/lib/Page.js#L421-L448) code.

Reply:
@aslushnikov - I tried reusing `page._onConsoleAPI` but I found that it does not make much sense:
 `_onConsoleAPI` initialize `ConsoleMessage`'s text from `args` while `Log.entryAdded` take the text from `event.text`. My question is if we do need `args` here at all? I can refactor `_onConsoleAPI` to be compatable for both cases (taking text from args or from event.text), but because I don't have a test case for `Log.entryAdded` with args, I cannot really tell if this use case is working.

WDYT?

Reply:
@yanivefraim I see. Let's just disregard `entryAdded` arguments. You can release them like console does.

https://github.com/GoogleChrome/puppeteer/blob/f797f8c307ff5380c750d3851c0b3d61a5299df5/lib/Page.js#L434-L437

I also figured there's no `executionContextId` in the `Log.entryAdded`, so you won't be able to convert these args to JSHandles anyway.

Reply:
@aslushnikov - thanks - fixed (I hope that I understood you correct)

BTW, I noticed `event.args.map(arg => helper.releaseObject(this._client, arg));` (in [here](https://github.com/GoogleChrome/puppeteer/blob/f797f8c307ff5380c750d3851c0b3d61a5299df5/lib/Page.js#L434-L437) does not wait for promise to be resolved. Is it intentionally?

Reply:
> BTW, I noticed event.args.map(arg => helper.releaseObject(this._client, arg)); (in here) does not wait for promise to be resolved. I assume it is because we do not care about the response? (just making sure)

That's right